deployer/src/deployment/run.rs

```diff
@@ -269,43 +277,191 @@ impl Built {
             .await
             .map_err(Error::Runtime)?;
 
-        kill_old_deployments.await?;
+        kill_old_deployments.await?; // TODO: Move to between load and provision? Or even after provision?
+
         // Execute loaded service
-        load(
+        let (old_resources, resources, new_secrets) = load(
             self.service_name.clone(),
             self.service_id,
-            executable_path.clone(),
-            resource_manager,
+            resource_manager.clone(),
             runtime_client.clone(),
-            self.claim,
+            self.claim.clone(),
             self.secrets,
         )
         .await?;
 
+        let resources = provision(
+            self.service_name.clone(),
+            self.service_id,
+            provisioner_client,
+            resource_manager,
+            self.claim,
+            old_resources,
+            resources,
+            new_secrets,
+        )
+        .await
+        .map_err(Error::Provision)?;
+
         let handler = tokio::spawn(run(
             self.id,
             self.service_name,
             runtime_client,
             address,
             cleanup,
+            resources,
         ));
 
         Ok(handler)
     }
 }
 
+async fn provision(
+    service_name: String,
+    service_id: Ulid,
+    mut provisioner_client: provisioner::Client,
+    mut resource_manager: impl ResourceManager,
+    claim: Claim,
+    old_resources: Vec<resource::Response>,
+    mut resources: Vec<Vec<u8>>,
+    new_secrets: HashMap<String, String>,
+) -> anyhow::Result<Vec<Vec<u8>>> {
+    let mut resources_to_save: Vec<record_request::Resource> = Vec::new();
+    for r in resources.iter_mut() {
+        match serde_json::from_slice::<ResourceInput>(r.as_slice())
+            .context("deserializing resource input")?
+        {
+            ResourceInput::Shuttle(shuttle_resource) => {
+                macro_rules! log {
+                    ($msg:expr) => {
+                        info!("[Resource][{}] {}", shuttle_resource.r#type, $msg);
+                    };
+                }
+                macro_rules! get_cached_output_or_provision {
```

Comment:
It's weird to have the macro rules here. Just move it to outside the function.

Reply:
For some reason it just did not work when I placed them there ...

Reply:
Neither works :skull: